### PR TITLE
update_adzerk: Only update flight data when a campaign is provided.

### PR DESCRIPTION
**Pull request by dwick**
Previously we were updating everything about the campaign,
advertiser, and creative when doing simple actions like pausing a
flight.  This should reduce the cost of updates.

:eyeglasses: @zeantsoi

### CODE REVIEW

File: reddit_adzerk/adzerkpromote.py [hunk 1]
```diff
@@ -482,7 +482,7 @@ def update_flight(link, campaign, az_campaign):
     return az_flight
 
 
-def create_cfmap(link, campaign, az_campaign, az_creative, az_flight):
+def create_cfmap(link, campaign):
```
Comment (zeantsoi) [inline reddit_adzerk/adzerkpromote.py]: `campaign` should be an optional defaulted to `None`, I think?

File: reddit_adzerk/adzerkpromote.py [hunk 2]
```diff
@@ -540,22 +540,19 @@ def _update_adzerk(link, campaign):
...
-        az_advertiser = update_advertiser(author)
-        az_campaign = update_campaign(link, az_advertiser)
-        az_creative = update_creative(link, az_advertiser)
 
-        if campaign:
-            az_flight = update_flight(link, campaign, az_campaign)
+        if campaign is None:
+            author = Account._byID(link.author_id, data=True)
+            az_advertiser = update_advertiser(author)
+            az_campaign = update_campaign(link, az_advertiser)
```
Comment (zeantsoi) [inline reddit_adzerk/adzerkpromote.py]: Is this needed?

File: reddit_adzerk/adzerkpromote.py (hunk 1 above)
Comment (zeantsoi) [inline reddit_adzerk/adzerkpromote.py]: Or, if not, where is `None` being assigned to the campaign being passed in? Separate commit?

Comment (dwick) [inline reddit_adzerk/adzerkpromote.py]: campaign is required here

File: reddit_adzerk/adzerkpromote.py (hunk 2 above)
Comment (dwick) [inline reddit_adzerk/adzerkpromote.py]: what needed?

File: reddit_adzerk/adzerkpromote.py (hunk 1 above)
Comment (zeantsoi) [inline reddit_adzerk/adzerkpromote.py]: In what instance would it be `None`?

File: reddit_adzerk/adzerkpromote.py (hunk 2 above)
Comment (zeantsoi) [inline reddit_adzerk/adzerkpromote.py]: `update_campaign`.

Comment (zeantsoi) [inline reddit_adzerk/adzerkpromote.py]: Nevermimd, mixed up the nomenclature.

File: reddit_adzerk/adzerkpromote.py (hunk 1 above)
Comment (dwick) [inline reddit_adzerk/adzerkpromote.py]: i think you mean for `update_adzerk`? https://github.com/reddit/reddit-plugin-adzerk/blob/30d5548253b3c79c11371592f3412055497e4841/reddit_adzerk/adzerkpromote.py#L609

File: reddit_adzerk/adzerkpromote.py (hunk 2 above)
Comment (dwick) [inline reddit_adzerk/adzerkpromote.py]: reddit link == adzerk campaign and creative
reddit campaign == adzerk flight and map

File: reddit_adzerk/adzerkpromote.py (hunk 1 above)
Comment (zeantsoi) [inline reddit_adzerk/adzerkpromote.py]: Whoops, you're right.

File: reddit_adzerk/adzerkpromote.py (hunk 2 above)
Comment (zeantsoi) [inline reddit_adzerk/adzerkpromote.py]: Just an idea, but being that we'll probably be adding some people over the next few months, maybe we should add some inline comments to this end.

**Comment by zeantsoi** [comment]
:fish:

**Pull request merged**